Gaming The Syllabus/Assets/Scripts/Riddles/UI/SyllabusRiddleUIManager.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+/// <summary>
+/// Singleton class responsible for displaying riddles by instantiating and initializing answer elements at runtime.
+/// Use <see cref="DisplayRiddle(Riddle)"/> to open the UI and display the passed in riddle. All logic required to
+/// support the user answering the riddle is handled appropriately by this class and those related to it.
+/// <para>Setup: Attach to Canvas and assign riddle uiPanel and question text</para>
+/// </summary>
+public class SyllabusRiddleUIManager : Singleton<SyllabusRiddleUIManager>
+{
+    [Header("UI Parameters")]
+    [SerializeField] private int maximumNumberOfAnswers = 4;
+    [SerializeField] private Vector2 firstAnswerPosition;
+    [SerializeField] private float yDistanceBetweenAnswerElements;
+
+
+    [Header("References")]
+    [SerializeField] private GameObject uiPanel;
+    [SerializeField] private TextMeshProUGUI question;
+    private List<IRiddleAnswerUIElement> riddleAnswerUIElements;
+
+    [Header("Answer Element Prefabs")]
+    [SerializeField] private GameObject buttonAnswerElementPrefab;
+
+    public static bool UIIsActive { get => _instance.uiPanel.activeInHierarchy; }
```

Comment:
Good call here on using activeInHierarchy. I previously have always used activeSelf, but activeInHiearchy is definitely a better approach.